go/libkb/keyfamily.go

```diff
@@ -1026,3 +1026,33 @@ func (ckf *ComputedKeyFamily) HasActiveDevice() bool {
 	}
 	return false
 }
+
+// Returns (&senderType, err). A non-nil error indicates some unexpected
+// condition (like the key doesn't exist at all), which should be propagated.
+// If the sender type is nil, the key is active, and the caller should proceed
+// with an identify. Otherwise the key is no longer active, and the sender type
+// indicates why.
+func (ckf ComputedKeyFamily) GetSaltpackSenderTypeIfInactive(kid keybase1.KID) (*keybase1.SaltpackSenderType, error) {
+	info := ckf.cki.Infos[kid]
+	if info == nil {
+		// This shouldn't happen without a server bug/attack or a very unlikely
+		// race condition (e.g. a user account reset between the API server
+		// telling us they own a key, and the loaduser confirming it.)
+		return nil, fmt.Errorf("Key %s not found in key infos", kid.String())
+	} else if info.Status == KeyRevoked {
```

Comment:
i'd say get rid if `else`s here if you're just returning out at the end of the `if`'s...